### ISSUE TITLE
Enable analytics on tool links on landing page

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-team#69057]
### User story

As a Product Manager in the MHV ecosystem, I need to be able to easily build reports on user flows that touch the MHV landing page ~and segment them by the versions of the landing page seen by different user types~, so that I can report on user journeys to stakeholders as well as use data to drive decisions about iterations on the landing page.
[NOTE: segmentation was split off to #71185 ]

### Notes

- [Comment from another ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/66723#issuecomment-1789106226) documents proposed schema that meets ACs
- "DataDog" in the front-end analytics context means DataDog RUM (real user monitoring), a suite of powerful tools for monitoring user interactions, e.g., heatmaps and session replays

## Acceptance criteria

- ~A custom user segment is available in Google Analytics ~~and in Datadog~~ for users who are authenticated but aren't registered at a facility~ _[moved to new issue #71185]_
- Events can be included in GA ~~and Datadog~~ reports for all elements of the page that contribute meaningfully to the UX of the page, including elements that display conditionally

## Tasks

- [ ]

**Comment by carlosfelix2** [comment]
Findings related to Google Analytics:
There is currently no user data being recorded that can identify a user as registered. However, we can add that data following GA's process. We would need to add a new custom dimension to GA, then populate the data in code accordingly. The proposed new custom dimension would be user scoped and should be set globally and not just for the landing page. We should think hard as to what data may be useful to try to go through this process once if possible. For comparison, other apps have added custom properties as well related to the user. For example, there are VAOS user specific properties and even a married flag that were added to GA, so I think adding our own should not be a problem. I also suspect that PII is not allowed, but that is not a problem for the data we need. 

As for process, I will suggest to consult with the analytics team before adding anything. Looks like property naming for something global is `Global - xxxx`.

**Comment by dcloud** [comment]
For `va-alert`, I looking into the [component library code](https://github.com/department-of-veterans-affairs/component-library/blob/b284b2100c8b329b710f12e54b4cf775955cc5cc/packages/web-components/src/components/va-alert/va-alert.tsx#L88C3-L88C21), and it seems that the alert automatically records an event when a link is clicked on inside the alert box, but not otherwise. My guess is we'd need to write an event handler for `va-component-did-load` and fire [a custom event](https://design.va.gov/about/developers/using-web-components#custom-events) that would call `recordEvent`. Unsure what key:values we'd pass to `recordEvent`, however

**Comment by carlosfelix2** [comment]
I looked more into how segments work per Wes' comments on Friday. Wes was right that we can use the segments to select those users that have visited a page or have an event, but now I have questions on how we can use that for our purposes:
If we have an event when the landing page was visited and the "no health alert" is shown, wouldn't that create a problem if that user gets registered later on and we are looking at user segments? I am thinking that we may be seeing page visits and clicks after the person registered when looking at a user like that.
With the same scenario of having an event when the landing page was visited and the "no health alert" is shown, I am thinking this can work to our advantage as we can use session segments and see what the user clicked on those sessions. There may be a session where the user was not registered and then it is, but I am thinking this will be a rare occurrence.
Now say I add a user scoped property/dimension that says if user is registered or not. I think this may create problems too when a non-registered user gets registered. I think once the user is registered then that user will no longer appear as part of a user segment for non-registered users and we can't see that data.
We can also add a session scoped property/dimension that says if the user is registered or not. If we do this globally and not just on the landing page we should be able to see what those users click on and perhaps we can change the site to drive traffic where we want to.
I do like #4. We can still add events when we load the landing page alert too. I think the event is easier to implement as a new property/dimension may require approval from the Analytics team. Thoughts?

**Comment by jzucadi** [comment]
This ticket is on hold until the analytics team completes [this submitted request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/70772) for tracking when a user is shown content that states they are not registered with a health facility

**Comment by wesrowe** [comment]
I scheduled a meeting Tues to discuss.

**Comment by wesrowe** [comment]
Moving this ticket to Review/QA, the blocked AC is its own ticket now.

**Comment by dcloud** [comment]
Yesterday I submitted a comment on this ticket, but apparently it didn't "take" so here I go again:

> the links at the bottom (R&S) are va-link components, which should get default analytics, but they have disable-analytics=true props in html. Did we disable it in order to fire our own events? (or were they disabled before we started the recent work?)

The links were originally plain html anchor `<a>` links, and based on being told we wanted to record custom dimensions, e.g. `destination-domain`, we need to fire our own event. We were also given the impression that plain `<a>` links were _not_ being recorded, which was a surprise to me.

> there were events being collected going months back??!! [GA report](https://analytics.google.com/analytics/web/#/report/content-event-pages/a50123418w177519031p176188361/_u.date00=20231201&_u.date01=20231201&explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.pagePath:www.va.gov~2Fmy-health~2F,analytics.eventCategory:Interactions/)

If events were being recorded, then the question now is: *Were we getting the information we need?* If our plain anchor links were being recorded, I'd think we'd know the destination url and wouldn't need a custom link component or a custom ga event.

**Comment by jzucadi** [comment]
[PR to remove destination domain and fix the object being sent back to Google Analytics instead of a string: ](https://github.com/department-of-veterans-affairs/vets-website/pull/26970)

**Comment by wesrowe** [comment]
I clicked _every single link_ in the six gray boxes, and AdSwerve showed the expected result in all cases.

One thing to notice for later – every expected event string ends in ` - (not set)`. I've noticed this in tons of cases all over VAgov, but thought I'd point it out here while it's still fresh in case anyone knows why that last bit is added on. 

It seems like it might be an opportunity. Is there maybe a parameter we could pass for the URL of the link?
